packages/astro/src/vite-plugin-markdown/index.ts

```diff
@@ -108,6 +108,9 @@ export default function markdown({ config }: AstroPluginOptions): Plugin {
 						Content.isAstroComponentFactory = true;
 						export function getHeaders() {
 							return load().then((m) => m.metadata.headers)
+						};
+						export function getSource() {
+							return load().then((m) => m.metadata.source)
```

Comment:
I wonder: do people want the source `Markdown`, or the compiled `Markdown -> HTML + Astro`? Or both? 

Throwing some ideas out here: If both, should we do one function for each (`getSource()` and `getResult()`) or one function that returns both (`getResult(): {md, html}`). 

Curious what you think!

Reply:
> do people want the source Markdown, or the compiled Markdown -> HTML + Astro?

For example, the raw source is used inside the docs ([this is my PR with a hacky implementation](https://github.com/withastro/docs/pull/296/files#diff-ef8db46d250278449d60bfbcecc6c06a5519cb716c0fec676335d2a5940664a8))

> Throwing some ideas out here: If both, should we do one function for each (getSource() and getResult()) or one function that returns both (getResult(): {md, html}).

Seems nice, but should we implement it when we have the `Content` component? Is there a case in which you would need the raw HTML that can't be replaced with `<Content />`?

Reply:
I'm not sure! can you remind me the thread where this request first came up? Lets see what the use-case was there and then design for that. We can always come back and clean it up later (in a non-breaking way, of course :)

Reply:
> can you remind me the thread where this request first came up?

There is no thread. When I was migrating the docs to Astro beta I've come up with this inconvenience. You can see the diff in the PR mention above. I can't use `<Content />` because it would result in a nested layout

Reply:
Oh, I see. Yeah, the html also works nice. I could make `getResult(): {md, html}`.

>  For ex, I'm not sure if it supports syntax highlighting, or the === set of features as a separate markdown file.

At least once my other PR gets merged, it should be almost identical!

Reply:
Hmm, maybe a convo for a different thread then! I think it's inevitable that `<Markdown>` support is degraded. Custom remark/rehype plugins come to mind, since those are provided via config and I don't think are bundled into your final SSR build.

Reply:
> I think it's inevitable that <Markdown> support is degraded

Yeah, it isn't the same. I now agree that having `getResult(): {md, html}` might be the best